lib/flipper/middleware/session.rb

```diff
@@ -7,9 +9,29 @@ def initialize(app, env_key: 'flipper')
 
   def call(env)
     original_instance = Flipper.instance
-    env[@env_key] = Flipper.instance = Flipper.new(Flipper::Adapters::Session.new(env['rack.session']))
+
+    env[@env_key] = Flipper.instance = self.class.configure(env['rack.session'])
+
+    # Sync before making request
+    Flipper.sync
+
     @app.call(env)
   ensure
     Flipper.instance = original_instance
   end
+
+  # Returns a Flipper instance using the given session
+  def self.configure(session, token: session[:cloud_token])
+    adapter = Flipper::Adapters::Session.new(session)
+
+    if token
+      Flipper::Cloud.new(
+        token: token,
+        local_adapter: adapter,
+        sync_secret: "disable polling"
```

Comment:
Seems like we need a better way to disable polling so this works into the future.